torch_geometric/explainability/algo/base.py

```diff
@@ -0,0 +1,168 @@
+        reference depending on the explainer mode (model vs phenomenon).
+
+        Args:
+            exp_output (torch.Tensor): the output of the explanation algorithm.
+                (e.g. the forward pass of the model with the mask applied).
+            gnn_output (torch.Tensor): the original output of the GNN.
+            target (torch.Tensor): the target of the GNN.
+            target_index (int): the index of the target to explain.
+        """
+
+    @abstractmethod
+    def loss(self, y_hat: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+        r"""
+        This method compute the loss to be used for the explanation algorithm.
+        Subclasses should override this method to define their own loss.
+
+        Args:
+            y_hat (torch.Tensor): the output of the explanation algorithm.
+                (e.g. the forward pass of the model with the mask applied).
+            y (torch.Tensor): the target output.
+        """
+
+    @property
+    def accept_new_loss(self) -> bool:
+        """Whether the algorithm can accept a new loss function."""
+        return True
+
+    def set_objective(
+        self,
+        objective: Callable[[torch.Tensor, torch.Tensor, torch.Tensor, int],
+                            torch.Tensor],
+    ) -> None:
+        """Sets the loss function to be used for the explanation algorithm.
+
+        Args:
+            objective (Callable): loss function.
+        """
+        self.objective = objective
+
+    @torch.no_grad()
+    def get_initial_prediction(
+        self,
+        model: torch.nn.Module,
+        g: Data,
+        batch: Optional[torch.Tensor] = None,
+        **kwargs,
+    ):
+        """Return the initial prediction of the model.
+
+        Args:
+            g (torch_geometric.data.Data): the input graph.
+            batch (torch.Tensor, optional): batch indicator. Defaults to None.
+
+        Returns:
+            torch.Tensor: output of the underlying model.
+        """
+        out = model(
+            g,
+            **dict(batch=batch, **kwargs),
+        )
+        return out
+
+    @abstractmethod
+    def explain(
+        self,
+        g: Data,
+        model: torch.nn.Module,
+        target: torch.Tensor,
+        target_index: Optional[int] = None,
+        batch: Optional[torch.Tensor] = None,
+        **kwargs,
+    ) -> Explanation:
+        """This method computes the explanation of the GNN.
+
+        Args:
+            g (Data): input graph.
+            model (torch.nn.Module): model to use in explanations.
+            target (torch.Tensor): target of the GNN.
+            target_index (int, optional): Index of the target to explain. Used
+                in case of multi-outputs. Defaults to None.
+            batch (torch.Tensor, optional): _description_. Defaults to None.
+
+        Returns:
+            Explanation: explanation of the GNN.
+        """
+
+    def forward(
```

Comment:
What's the purpose of `explain` when we can directly use `forward` for this?

Reply:
Brain on autopilot, will update.